src/nix/buildpack/tools/nix.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -e
+
+check_semver ${NIX_VERSION}
+
+if [[ ! "${MAJOR}" || ! "${MINOR}" || ! "${PATCH}" ]]; then
+  echo Invalid version: ${NIX_VERSION}
+  exit 1
+fi
+
+echo "max-jobs = auto" | tee -a /tmp/nix.conf >/dev/null
+echo "trusted-users = root ubuntu" | tee -a /tmp/nix.conf >/dev/null
+
+installer_options=(
+  --nix-extra-conf-file /tmp/nix.conf
+)
```

Comment:
I saw some more install steps, should we run the optimization steps to minimize layer size?

https://github.com/NixOS/docker/blob/e88fd1a6df223b831481683a0ac773656ddb90fc/Dockerfile#L11-L23

I think we also need `echo 'sandbox = false' > /etc/nix/nix.conf` because user ubuntu won't be able to swich to the nix user?

Reply:
I've added the optimisation, but not sure if `sandbox = false` is required. Added it anyway.